keep-ui/app/rules/AIGenRules.tsx

```diff
@@ -0,0 +1,318 @@
+
+  const [loadingRows, setLoadingRows] = useState<{ [key: string]: boolean }>({});
+  const [successRows, setSuccessRows] = useState<{ [key: string]: boolean }>({});
+
+  const mutateAIGeneratedRules = (rules: AIGeneratedRule[]) => {
+    setGeneratedRules(rules);
+    setIsLoadingRules(false);
+  };
+
+  const { data:serverGenRules } = useRulePusherUpdates();
+
+  useEffect(() => {
+    console.log("serverGenRules", serverGenRules);
+
+    if (Array.isArray(serverGenRules) && (0 === serverGenRules.length)) {
+      return;
+    }
+    
+    mutateAIGeneratedRules(serverGenRules);
+  }, [serverGenRules]);
+
+  const handleGenerateMoreRules = () => {
+    setIsLoadingRules(true);
+    triggerGenRules();
+  };
+
+  const handleAddRule = async (rule: AIGeneratedRule) => {
+    const ruleKey = rule.ShortRuleName;
+    setLoadingRows((prev) => ({ ...prev, [ruleKey]: true }));
+    try {
+      const temp = {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'Authorization': `Bearer ${session?.accessToken}`,
+        },
+        body: JSON.stringify({
+          ruleName: rule.ShortRuleName,
+          sqlQuery: { sql: '{new-version-not-adding-this}', params: ['no-params'] },
+          celQuery: rule.CELRule,
+          timeframeInSeconds: rule.Timeframe,
+          timeUnit: 'minutes',
+          groupingCriteria: rule.GroupBy,
+          groupDescription: rule.ChainOfThought,
+          requireApprove: false,
+        }),
+      };
+
+      const apiUrl = getApiURL();
+      const response = await fetch(`${apiUrl}/rules`, temp);
+
+      if (response.ok) {
+        setSuccessRows((prev) => ({ ...prev, [ruleKey]: true }));
+        mutate(`${apiUrl}/rules`); // Refresh the rules list
+      } else {
+        const result = await response.json();
+        console.error('Failed to add rule:', result);
+      }
+    } catch (error) {
+      console.error('Error adding rule:', error);
+    } finally {
+      setLoadingRows((prev) => {
+        const newLoadingRows = { ...prev };
+        delete newLoadingRows[ruleKey];
+        return newLoadingRows;
+      });
+    }
+  };
+
+  const columns = React.useMemo(() => [
+    columnHelper.accessor("ShortRuleName", {
+      header: "Short Rule Name",
+    }),
+    columnHelper.accessor("Score", {
+      header: "Score",
+    }),
+    columnHelper.accessor("CELRule", {
+      header: "CEL Rule",
+      cell: (info) => <div className="text-wrap">{info.getValue()}</div>,
+    }),
+    columnHelper.accessor("Timeframe", {
+      header: "Timeframe",
+    }),
+    columnHelper.accessor("GroupBy", {
+      header: "Group By",
+      cell: (info) => info.getValue().join(", "),
+    }),
+    columnHelper.accessor("ChainOfThought", {
+      header: "Explanations",
+      cell: (info) => {
+        const rule = info.row.original;
+        return (
+          <div className="flex space-x-2">
+            <Icon
```

Comment:
to fix tooltip, try to add
```suggestion
            <Icon
             key={rule.ChainOfThought}
```

my guess is that at first render content is not yet here but dimensions are already set.